### ISSUE TITLE
Define structure for different message types

### ISSUE THREAD

**Issue by haardikk21** [Ryan-Ouyang/metamask-messaging#10]
Currently all messages are considered text. We should consider and support other types of messages such as transactions (a signed tx blob), invoices (money requests waiting to be signed), receipts (??) and potentially more.

**Comment by jsdu** [comment]
We should also keep in mind for different languages and emojis

**Comment by Ryan-Ouyang** [comment]
We started out saying that any arbitrary data was allowed since we were using IPFS - I agree that we should have multiple predefined types, but also make it possible for arbitrary data blobs to be sent since IPFS allows us that flexibility.

**Comment by haardikk21** [comment]
> We started out saying that any arbitrary data was allowed since we were using IPFS - I agree that we should have multiple predefined types, but also make it possible for arbitrary data blobs to be sent since IPFS allows us that flexibility.

Yes. We should look into binary encoding using msgpack or something similar for everything in IPFS actually instead of sending stuff as text.
